ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -376,6 +376,27 @@ public boolean updateBlobExpiration(BlobId blobId, long expirationTime) throws C
     }
   }
 
+  /**
+   * Filter out {@link CloudBlobMetadata} objects from lastUpdateTime ordered {@code cloudBlobMetadataList} whose
+   * lastUpdateTime is {@code lastUpdateTime} and id is in {@code lastReadBlobIds}.
+   * @param cloudBlobMetadataList list of {@link CloudBlobMetadata} objects to filter out from.
+   * @param lastReadBlobIds set if blobIds which need to be filtered out.
+   * @param lastUpdateTime lastUpdateTime of the blobIds to filter out.
+   */
+  private void filterOutLastReadBlobs(List<CloudBlobMetadata> cloudBlobMetadataList, Set<String> lastReadBlobIds,
+      long lastUpdateTime) {
+    List<CloudBlobMetadata> removeList = new LinkedList<>();
+    for (CloudBlobMetadata cloudBlobMetadata : cloudBlobMetadataList) {
```

Comment:
Minor: you can actually iterate through the input list and call iter.remove() along the way (a bit more efficient).